CONTRIBUTING.MD

```diff
@@ -0,0 +1,135 @@
+# Contributing to Kogito Operator
+
+Many thanks for your interest in contributing to this project. We love contributions! :heart:
+
+Any kind of contribution is welcome: code, design ideas, bug reporting, documentation (including this one!) and so on.
+
+## Some tips before you start
+
+If you understand that you can contribute with code, here's some advice that might help during your journey:
+
+1. Make yourself comfortable with Go language. If you're a new programmer, take a look at the [official docs](https://golang.org/doc/). For more advanced/intermediate level, [The Go Programming Language](https://www.amazon.com/gp/product/0134190440/) book might be a good starting point.
+2. Become familiar with [the Operator SDK](https://github.com/operator-framework/operator-sdk). They have a nice [user guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md) that you can follow using [Memcached Operator](https://github.com/operator-framework/operator-sdk-samples/tree/master/memcached-operator) as an example.
+3. Make sure to have all [requirements](README.md#requirements) set on your local machine.
+
+## Architecture
+
+Kogito Operator was designed to deploy [Kogito Runtimes](https://github.com/kiegroup/kogito-runtimes) services from source and every piece of infrastructure that it might need like SSO ([Keycloak](https://github.com/integr8ly/keycloak-operator)) and Persistence ([Infinispan](https://github.com/infinispan/infinispan-operator)).
+
+The actual architecture has only one [controller](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg#hdr-Controller) that's responsible to deploy the application from source. It's on the [roadmap](https://github.com/kiegroup/kogito-runtimes/wiki/Roadmap) to have two more controllers to handle SSO and Persistence. The following image illustrates the general idea:
+
+![Kogito Operator General Architecture](docs/img/general_archictecture.png?raw=true)
+
+One of the most important responsibilities of the controller is the [Reconcile Loop](https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#reconcile-loop). Inside this "loop" the controller will ensure that it has every resource (Kubernetes and OpenShift objects) created and updated accordingly.
+
+We aim to avoid having a huge monolithic inside the reconcile loop that does it all. With that in mind, we separated the responsibility of the Kubernetes and OpenShift API calls to a package that we call [`inventory`](pkg/controller/kogitoapp/inventory) and the definition of the resources (our domain) into the [`definitions`](pkg/controller/kogitoapp/definitions) package.
```

Comment:
"having a huge monolithic inside" - a word is missing between monolithic and inside. If "monolithic" was supposed to be a noun, please replace with "monolith".

Reply:
"responsibility of the Kubernetes and OpenShift API calls" is not clear at all to me. Perhaps of "making" the calls?